tool/src/js.rs

```diff
@@ -68,106 +68,130 @@ fn gen_method<W: fmt::Write>(
 ) -> fmt::Result {
     // TODO(shadaj): support results with empty success value
     // TODO(shadaj): reconsider if we should auto-detect writeables
-    if method.return_type.is_none()
+    let is_writeable = method.return_type.is_none()
```

Comment:
suggestion: I suspect `method` may benefit from a `has_writeable_input` method that panics when there is a return type.